assets/prometheus-k8s/rules.yaml

```diff
@@ -334,6 +334,18 @@ spec:
     - expr: sum(rate(container_cpu_usage_seconds_total{container_name!="POD",container_name!="",pod_name!=""}[5m]))
         / sum(machine_cpu_cores)
       record: cluster:container_cpu_usage:ratio
+    - expr: sum({__name__="machine_cpu_cores"})
+      record: cluster_cpu_cores
+    - expr: sum({__name__="node:node_cpu_utilisation:avg1m"})
+      record: cluster_cpu_utilisation:avg1m
+    - expr: sum({__name__="machine_memory_bytes"})
+      record: cluster_memory_bytes
+    - expr: sum({__name__="node:node_memory_utilisation:ratio"})
+      record: cluster_memory_utilisation:ratio
+    - expr: sum({__name__="node_filesystem_size_bytes"})
```

Comment:
This heavily relies on the underlying metrics to only be collected from physical filesystems, which I don't think is true. Could you share an example of the time-series returned by this metric?

Reply:
@brancz, this metrics was chosen because this is the metrics uesd in Grafana dahsboard [1].

[1] https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/master/dashboards/use.libsonnet#L79

> Could you share an example of the time-series returned by this metric?

This example show size vs. free per node:
`{__name__=~"node_filesystem_free_bytes|node_filesystem_size_bytes"}`
This example show cluster:
`sum({__name__=~"node_filesystem_free_bytes"})`

Reply:
@brancz we are not set on what metrics to use here, if you think of a better metric to show for cluster `storage` totals and utilization I will be happy to change this metrics to something that will be more helpful to end users.

Reply:
That's the metric name used, but the whole query we end up using is:

```
sum(max(node_filesystem_size_bytes{fstype=~"ext[234]|btrfs|xfs|zfs"} - node_filesystem_avail_bytes{fstype=~"ext[234]|btrfs|xfs|zfs"}) by (device,pod,namespace)) by (pod,namespace)
/ scalar(sum(max(node_filesystem_size_bytes{fstype=~"ext[234]|btrfs|xfs|zfs"}) by (device,pod,namespace)))
* on (namespace, pod) group_left (node) node_namespace_pod:kube_pod_info:
```